gateway/mw_upstream_basic_auth_ee.go

```diff
@@ -0,0 +1,14 @@
+//go:build ee || dev
+
+package gateway
+
+import (
+	"github.com/TykTechnologies/tyk/ee/middleware/upstreambasicauth"
+)
+
+func getUpstreamBasicAuthMw(base *BaseMiddleware) TykMiddleware {
+	spec := base.Spec
+	mwSpec := upstreambasicauth.NewAPISpec(spec.APIID, spec.Name, spec.IsOAS, spec.OAS, spec.UpstreamAuth)
```

Comment:
```suggestion
	mwSpec := spec.Model()
```

in line with a copy of the full APISpec in internal/model. Can use [huandu/go-clone](https://github.com/huandu/go-clone).

There might be a bridge already with model.MergedAPIs as apidef and oas objects are out of gateway scope. Meaning, you could take a model.MergedAPI today without the APISpec copy

Reply:
this is going away, resolving